app/components/sy-modal/overlay/template.hbs

```diff
@@ -1 +1,6 @@
-{{yield}}
+<div
+  class="modal modal-overlay {{if @visible 'modal--visible'}}"
+  {{did-insert this.setupClickHandler}}
```

Comment:
Why can't we use something like `{{ on "click" this.handleClick }}`?

Reply:
We need the components wrapping HTML-`element` to compare it in the on-click handler.